protocol/contracts/oud/OudRedeemer.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Oud Redeemer (protocol/contracts/oud/OudRedeemer.sol)
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '../interfaces/oud/IOudRedeemer.sol';
+import '../interfaces/oud/Oud__ITempleLineOfCredit.sol';
+import '../interfaces/oud/IOudtoken.sol';
+import '../interfaces/oud/Oud__ITempleERC20Token.sol';
+
+/**
+ * @title Allows for the redemption of Oud + a stable coin for Temple at the Temple Treasury Price index
+ */
+contract OudRedeemer is IOudRedeemer, Ownable {
+  using SafeERC20 for IERC20;
+  error AddressZero();
+  error RedeemAmountZero();
+
+  // @notice Temple token address
```

Comment:
Comments need fixing for natspec

Reply:
[a887d99](https://github.com/TempleDAO/temple/pull/760/commits/a887d990cd0aa465cc7c5237e60d4e7a1f580dc1)